src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -501,6 +501,13 @@ public class Options
             [Argument(ArgumentType.AtMostOnce, HelpText = "Use separate parsing threads?", ShortName = "threads", Hide = true)]
             public bool UseThreads = true;
 
+            /// <summary>
+            /// If true, new line characters are acceptable inside a quoted field, and thus one field can have multiple lines of text inside it
+            /// If <see cref="TextLoader.Options.AllowQuoting"/> is false, this option is ignored.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Escape new line characters inside a quoted field? If AllowQuoting is false, this argument is ignored.", ShortName = "multilines", Hide = true)]
```

Comment:
Can it share the same value of `AllowQuoting`? The thought behind it is if `AllowQuoting` is `True`, we  also need  `ReadMultilines` to be `True` to make sure `GetSomeLines` behaves correctly

Reply:
Unrelated questions, why we need to add attribute to those text loader options. I dig into a bit and it seems to be exposed to a command line tool, what's that tool?

Reply:
Yeah. So we can do that, but I don't know if we should (so I'd like other people's feedback on this as well).

The thing is that if we always enable reading multilines when AllowQuoting is true, then we're risking ourselves to 2 problems:
1. Not reading multiline will be faster and reading it, because there are some extra steps that need to be taken when using multilines. I haven't ran benchmarks, so I don't know how much slower this would be.
2. If the CSV has invalid format, particularly if it has a quoted field that opens a " that it never closes, then if readMultiline is enabled, then all of the file is loaded into memory and assigned to that line (I mention it [here](https://github.com/dotnet/machinelearning/pull/5125#discussion_r425034529)). In contrast, if we don't allow readMultiline but we have AllowQuoting = true, then the current behavior is that TextLoader will load that line, and when parsing it it will log that it has a quoted field that opens but never closes, and will return it as an empty field. So currently ML.NET will pretty much ignore this badly formatted line, but we wouldn't be able to do this if we set readMultiline to true.

Reply:
> Unrelated questions, why we need to add attribute to those text loader options. I dig into a bit and it seems to be exposed to a command line tool, what's that tool?

My understanding is that they're there to support a legacy tool called MAML. Back in TLC (the predecessor of ML.NET) users could run MAML in the command line to build and run pipelines. MAML isn't publicly advertised for ML.NET, and we don't expect users to use it (although we still build the tool, and we use it in several internal tests). Also, don't quote me on this, but _I think_ that the descriptions given in those attributes are used to generate a manifest.json that contains descriptions of pretty much all publicly available components. This manifest.json is then used by other tools such as NimbusML (the python wrapper for ML.NET) to check for updates on ML.NET's API and correctly interface with it.

Reply:
### Resolved